lib/elixir/lib/code/identifier.ex

```diff
@@ -105,8 +105,34 @@ defmodule Code.Identifier do
 
   defp escape_char(0), do: [?\\, ?0]
 
-  @escaped_bom :binary.bin_to_list("\\uFEFF")
-  defp escape_char(65279), do: @escaped_bom
+  defp escape_char(char)
+       # Some characters that are confusing (zero-width / alternative spaces) are displayed
+       # using their unicode representation:
+       # https://en.wikipedia.org/wiki/Universal_Character_Set_characters#Special-purpose_characters
+
+       # BOM
+       when char == 0xFEFF
+       # Mathematical invisibles
+       when char in 0x2061..0x2064
+       # Bidirectional neutral
+       when char in [0x061C, 0x200E, 0x200F]
+       # Bidirectional general (source of vulnerabilities)
+       when char in 0x202A..0x202E
+       when char in 0x2066..0x2069
```

Comment:
I think it's especially good for this one, since we already have something to prevent vulnerabilities around it in the tokenizer: https://github.com/elixir-lang/elixir/blob/e8ea6a52596324cef9ebb95390d51c84c93bf73d/lib/elixir/src/elixir_tokenizer.hrl#L22-L24